packages/react/src/RendererContext.tsx

```diff
@@ -28,18 +28,13 @@ const RendererContext = React.createContext<GriffelRenderer>(createDOMRenderer()
  * @public
  */
 export const RendererProvider: React.FC<RendererProviderProps> = ({ children, renderer, targetDocument }) => {
-  'use no memo';
-
-  if (canUseDOM()) {
-    // This if statement technically breaks the rules of hooks, but is safe because the condition never changes after
-    // mounting.
-    // eslint-disable-next-line react-hooks/rules-of-hooks
-    React.useMemo(() => {
+  React.useMemo(() => {
```

Comment:
Actually, it could be a nice fit for `useInsertionEffect()`. @spmonahan if you interested, feel free to take a look 😉

Reply:
Yeah, looks like a possibility :)

My focus on React Compiler changes so far has been to just grant any existing code an exception so we can lint without errors and then come back and revisit all the exceptions later. Basically, don't break anything that is currently working ;)